code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -359,36 +364,111 @@
+		pressure_difference *= -1 // We want this positive
+		pressure_direction = turn(pressure_direction, 180) // Flip er around
+
+	archived_pressure_difference = pressure_difference
+	archived_pressure_direction = pressure_direction
+	pressure_difference = 0
+
+	var/force = sqrt(archived_pressure_difference) * (MOVE_FORCE_DEFAULT / 5)
+	for(var/atom/movable/moving_atom as anything in src)
+		if(moving_atom.pressure_moving || moving_atom.anchored)
+			continue
+		var/pressure_ratio = GET_PRESSURE_RATIO(moving_atom, src)
+		// Need at least enough pressure to move us
+		if(pressure_ratio > 1)
+			continue
+
+		if(force <= (moving_atom.move_resist * MOVE_FORCE_PUSH_RATIO * 0.5))
+			continue
+		moving_atom.start_pressure_move(pressure_ratio, archived_pressure_direction)
 
 /atom/movable
-	///How much delta pressure is needed for us to move
+	/// How much pressure is needed for us to move. Also effects rate of movement.
 	var/pressure_resistance = 10
-	var/last_high_pressure_movement_air_cycle = 0
+	/// Is this atom currently being moved by pressure?
+	var/pressure_moving = FALSE // Yes we could use a proc call for this, but it's hot, so we're using a var
+	/// Does this atom want to be asked in get_pressure_from about the amount of force acting on it?
+	/// Or does it just use pressure_resistance
+	var/complex_pressure = FALSE
+
+/atom/movable/proc/start_pressure_move(pressure_ratio, direction, added_flags = MOVEMENT_LOOP_START_FAST)
+	pressure_moving = TRUE
+	var/rate = LERP(PRESSURE_DELAY_MIN, PRESSURE_DELAY_MAX, pressure_ratio) // If pressure ratio's 0, we want MIN, if it's 1 we want MAX. Much nicer then a clamp
+	var/datum/move_loop/loop = SSmove_manager.move(src, direction = direction, delay = rate, subsystem = SSpressure_movement, flags = MOVEMENT_LOOP_IGNORE_PRIORITY|added_flags)
+	RegisterSignal(loop, COMSIG_MOVELOOP_PREPROCESS_CHECK, /atom/movable/proc/pre_pressure_move)
+	RegisterSignal(loop, COMSIG_PARENT_QDELETING, /atom/movable/proc/release_pressure)
+
+/atom/movable/proc/pre_pressure_move(datum/move_loop/move/source)
+	SIGNAL_HANDLER
+	if(anchored || !istype(loc, /turf/open))
+		qdel(source)
+		return MOVELOOP_SKIP_STEP
+
+	var/turf/open/our_turf = loc
+
+	// Need at least enough pressure to move us. We also make the assumption that pressure resist won't change
+	if(GET_PRESSURE_RATIO(src, our_turf) > 1)
+		qdel(source)
+		return MOVELOOP_SKIP_STEP
+
+	var/force = sqrt(our_turf.archived_pressure_difference) * (MOVE_FORCE_DEFAULT / 5)
+	if(force <= (move_resist * MOVE_FORCE_PUSH_RATIO * 0.5))
+		qdel(source)
+		return MOVELOOP_SKIP_STEP
+
+	if(SEND_SIGNAL(src, COMSIG_ATOM_PRE_PRESSURE_PUSH) & COMSIG_ATOM_BLOCKS_PRESSURE || pulledby)
+		return MOVELOOP_SKIP_STEP
+
+	source.direction = our_turf.archived_pressure_direction
+
+/atom/movable/proc/release_pressure(datum/move_loop/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(src, COMSIG_MOVABLE_MOVED)
+	pressure_moving = FALSE
+
+/atom/movable/Moved(atom/OldLoc, Dir)
```

Comment:
This is a double definition that makes all movements incur an additional proc call overhead.